addons/website/static/src/client_actions/configurator/configurator.js

```diff
@@ -442,47 +443,101 @@ export class FeaturesSelectionScreen extends ApplyConfiguratorScreen {
-                    }));
-                }
-                this.themeSVGPreviews[idx].el.appendChild(svgEl);
-            });
-            // When all the images inside the svgs are loaded then remove the
-            // loading effect.
-            Promise.allSettled(proms).then(() => {
-                this.uiService.unblock();
-            });
+        onMounted(() => {
+            this.blockUiDuringImageLoading(this.state.themes, this.themeSVGPreviews);
+        });
+    }
+
+    /**
+     * Transforms text svgs into svg elements and adds a loading effect that
+     * blocks the UI during the loading of the images inside those svg elements.
+     * @param {Array<Object>} themes - The text svgs.
+     * @param {Array} themeSVGPreviews - A reference to the svg elements.
+     */
+    blockUiDuringImageLoading(themes, themeSVGPreviews) {
+        const proms = [];
+        this.uiService.block();
+        themes.forEach((theme, idx) => {
+            const svgEl = new DOMParser().parseFromString(theme.svg, "image/svg+xml").documentElement;
+            for (const imgEl of svgEl.querySelectorAll("image")) {
+                proms.push(new Promise((resolve, reject) => {
+                    imgEl.addEventListener("load", () => {
+                        resolve(imgEl);
+                    }, {once: true});
+                    imgEl.addEventListener("error", () => {
+                        reject(imgEl);
+                    }, {once: true});
+                }));
+            }
+            themeSVGPreviews[idx].el.appendChild(svgEl);
+        });
+        // When all the images inside the svgs are loaded then remove the
+        // loading effect.
+        Promise.allSettled(proms).then(() => {
+            this.uiService.unblock();
         });
     }
 
     async chooseTheme(themeName) {
         await this.applyConfigurator(themeName);
     }
+
+    async getMoreThemes() {
+        const removeLoadingEffect = addButtonLoadingEffect(this.extraThemesButtonRef.el);
+        const themes = await this.orm.call("website",
+            "configurator_recommended_themes",
+            [],
+            {
+                "industry_id": this.state.selectedIndustry.id,
+                "palette": this.state.selectedPalette,
+                "result_nbr_max": this.nbrMainThemes + this.nbrExtraThemes,
+            },
+        );
+        // Filter the extra themes to not propose a theme that is already
+        // present in the main themes.
+        const mainThemeNames = this.state.themes.map((theme) => theme.name);
+        const extraThemes = themes.filter((extraTheme) => !mainThemeNames.includes(extraTheme.name));
+        this.state.updateExtraThemes(extraThemes, this.nbrExtraThemes);
+        removeLoadingEffect();
+        if (!this.state.extraThemes.length) {
+            // Hide the extra themes button as no extra themes have been found
+            this.extraThemesButtonRef.el.classList.add("d-none");
+        }
```

Comment:
That info should be in the state, so that this button is not even rendered in the case no extra theme was found

Reply:
Adapted but I am not sure if the way I did is the best: Indeed, I modified `env.store` to add the extra `showExtraThemesButton` element to the state. I did it this way as `env.store` is a proxy (and so, I can not do something like `Object.assign({}, env.store, {'showExtraThemesButton':true})`). The problem is that I modify the store so components that use the store see `showExtraThemesButton` that is only relevant for the `ThemeSelectionScreen` component.